src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8632,6 +8632,18 @@ public override BoundNode VisitEventAssignmentOperator(BoundEventAssignmentOpera
             }
             VisitRvalue(node.Argument);
             // https://github.com/dotnet/roslyn/issues/31018: Check for delegate mismatch.
+            if (node.Argument.ConstantValue?.IsNull != true
+                && MakeMemberSlot(receiverOpt, @event) is not -1 and var memberSlot)
+            {
+                if (node.IsAddition)
+                {
+                    this.State[memberSlot] = this.State[memberSlot].Meet(ResultType.State);
+                }
+                else
+                {
+                    this.State[memberSlot] = NullableFlowState.MaybeNull;
```

Comment:
I decided to change these to more direct state modifications because event assignment is more limited in what it can operate on. For example the LHS of an event assignment must be an lvalue, so it can't be a conditional-access expression like `x?.y += SomeDelegate`.

If this feels wrong from a coding patterns POV we can change this to use the LearnFromNullTest/LearnFromNonNullTest or some other more appropriate helpers that I may have overlooked in NullableWalker.